xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go

```diff
@@ -302,3 +313,217 @@ func (s) TestErrorFromParentLB_ResourceNotFound(t *testing.T) {
 		t.Fatalf("RPCs did not fail after removal of Cluster resource")
 	}
 }
+
+// testChildPolicyBalancerBuilder wraps the priority LB policy (which is the
+// child policy for the cluster resolver LB policy), and makes certain events
+// available to the test (subConn state changes and config pushes).
```

Comment:
Nit: instead of config pushes (which I assume to come from UpdateClientConnState) can you instead say ParseConfig(). I know it *should* come 1:1 (ParseConfig than UpdateClientConnState) but I had to fix an issue in cds where we were bypassing this API here: https://github.com/grpc/grpc-go/pull/6361/files#diff-20fc9eddb97bc2324ff2b3ca5bd2cada8c7f4ad23776d547ad08e09087699bb0R376.